app/parse.py

```diff
@@ -8,8 +16,73 @@ class Quote:
     tags: list[str]
 
 
+def parse_single_quote(quote_soup: BeautifulSoup) -> Quote:
+    return Quote(
+        text=quote_soup.select_one("span.text").text,
+        author=quote_soup.select_one("small.author").text,
+        tags=[tag.text for tag in quote_soup.select("a.tag")],
+    )
+
+
+def fetch_page_content(page_url: str) -> bytes | None:
+    try:
+        response = requests.get(page_url)
+        response.raise_for_status()
+
+        return response.content
+
+    except requests.HTTPError:
+        return None
```

Comment:
It is better to provide the error message in this case insted of returning None ) 
All good overall